metricflow/sql_clients/big_query.py

```diff
@@ -13,7 +13,7 @@
 from metricflow.sql_clients.sqlalchemy_dialect import SqlAlchemySqlClient
 
 
-class BigQueryEngineAttributes:
+class BigQueryEngineAttributes(SqlEngineAttributes):
```

Comment:
As a general rule we should not inherit from protocols, and we should NEVER inherit from protocols that define bare properties (such as `SqlEngineAttributes`) as this defeats the purpose of using a protocol as a type definition.

With this change, if you left this class empty it'd pass the typechecker and then blow up in the runtime.

Reply:
In that case, how can we improve the developer experience? E.g. I want to add an attribute to the protocol, and I need to find all classes that need updates.

Reply:
I generally add the attribute to the protocol and then run mypy (or find new mypy errors in the IDE) if I don't already know what I'm looking for. This can lead to a big fanout explosion of errors, but it's easy enough to tame those.